src/crawler.ts

```diff
@@ -1295,36 +1300,50 @@ self.__bx_behaviors.selectMainBehavior();
     await this.postCrawl();
   }
 
+  async closePages() {
+    if (this.pagesFH) {
+      try {
+        await new Promise<void>((resolve) =>
+          this.pagesFH!.close(() => resolve()),
+        );
+      } catch (e) {
+        // ignore
+      } finally {
+        this.pagesFH = null;
+      }
+    }
+
+    if (this.extraPagesFH) {
+      try {
+        await new Promise<void>((resolve) =>
+          this.extraPagesFH!.close(() => resolve()),
+        );
+      } catch (e) {
+        // ignore
+      } finally {
+        this.extraPagesFH = null;
+      }
+    }
+  }
+
   async postCrawl() {
     if (this.params.combineWARC) {
       await this.combineWARC();
     }
 
     if (this.params.generateCDX) {
+      // just move cdx files from tmp-cdx -> indexes at this point
       logger.info("Generating CDX");
-      await fsp.mkdir(path.join(this.collDir, "indexes"), { recursive: true });
-      await this.crawlState.setStatus("generate-cdx");
 
-      const warcList = await fsp.readdir(path.join(this.collDir, "archive"));
-      const warcListFull = warcList.map((filename) =>
-        path.join(this.collDir, "archive", filename),
-      );
+      const src = path.join(this.collDir, "tmp-cdx");
+      const dest = path.join(this.collDir, "indexes");
 
-      //const indexResult = await this.awaitProcess(child_process.spawn("wb-manager", ["reindex", this.params.collection], {cwd: this.params.cwd}));
-      const params = [
-        "-o",
-        path.join(this.collDir, "indexes", "index.cdxj"),
-        ...warcListFull,
-      ];
-      const indexResult = await this.awaitProcess(
-        child_process.spawn("cdxj-indexer", params, { cwd: this.params.cwd }),
-      );
-      if (indexResult === 0) {
-        logger.debug("Indexing complete, CDX successfully created");
-      } else {
-        logger.error("Error indexing and generating CDX", {
-          "status code": indexResult,
-        });
+      await this.crawlState.setStatus("generate-cdx");
+      await fsp.mkdir(dest, { recursive: true });
+
+      const tmpCdxList = await fsp.readdir(src);
+      for (const filename of tmpCdxList) {
+        await fsp.copyFile(path.join(src, filename), path.join(dest, filename));
```

Comment:
This is a fairly significant change in outputs, from a single `index.cdjx` file or it's gzipped equivalent to a number of CDXJ files named `<warc-file>.warc.gz.cdx`.

1. We should consider whether there are ramifications for this, including if the CDXJ files are large enough that they should be gzipped, and whether it'll be a breaking change for any users.
2. Since the tmp-cdx files being copied over are actually CDXJ, we should use the more accurate `.cdxj` extension for these when writing them

Reply:
Resolved by using warcio-js's `CDXIndexer` to reindex from WARCS if `--generateCDX` is passed.